redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -294,83 +294,263 @@ inline void getValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
 }
 
 /**
- * Systems derived class for delivering OemPowerSupplyMetrics Schema.
+ * @brief Retrieves valid input history item.
+ *
+ * Not all power supplies support the power input history. Do not provid Redfish
+ * fields for input power history if no associated endpoint matches this
+ * chassis.
+ *
+ * @param asyncResp   Pointer to object holding response data
+ * @param chassisID
+ * @param powerSupplyID
+ * @param callback  Callback for next step to get valid chassis ID
  */
-inline void requestRoutesPowerSupplyMetrics(App& app)
+template <typename Callback>
+inline void
+    getValidInputHistory(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& chassisID,
+                         const std::string& powerSupplyID, Callback&& callback)
 {
-    BMCWEB_ROUTE(app, "/redfish/v1/Chassis/<str>/PowerSubsystem/"
-                      "PowerSupplies/<str>/Metrics")
-        .privileges({{"Login"}})
-        .methods(boost::beast::http::verb::get)(
-            [](const crow::Request&,
-               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-               const std::string& chassisID, const std::string& powerSupplyID) {
-                auto getChassisID = [asyncResp, chassisID, powerSupplyID](
-                                        const std::optional<std::string>&
-                                            validChassisID) {
-                    if (!validChassisID)
+    BMCWEB_LOG_DEBUG << "getValidInputHistory enter";
+
+    const std::array<const char*, 2> interfaces = {
+        "org.open_power.Sensor.Aggregation.History.Average",
+        "org.open_power.Sensor.Aggregation.History.Maximum"};
+
+    auto respHandler = [callback{std::move(callback)}, asyncResp, chassisID,
+                        powerSupplyID](
+                           const boost::system::error_code ec,
+                           const std::vector<std::pair<
+                               std::string,
+                               std::vector<std::pair<
+                                   std::string, std::vector<std::string>>>>>&
+                               subtree) {
+        BMCWEB_LOG_DEBUG << "getValidInputHistory respHandler enter";
+
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "getValidInputHistory respHandler DBUS error: "
+                             << ec;
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        // Set the default value to resourceNotFound, and if we confirm finding
+        // association with chanssisID and powerSupplyID is correct, the error
```

Comment:
Nit: spelling

Reply:
That was tricky to find the word spelled wrong. I might need a tool to find and fix these. :-)